sdk/ml/azure-ai-ml/azure/ai/ml/_azure_environments.py

```diff
@@ -189,3 +198,46 @@ def _resource_to_scopes(resource):
     """
     scope = resource + "/.default"
     return [scope]
+
+def _get_clouds_by_metadata_url(metadata_url, timeout=ArmConstants.DEFAULT_TIMEOUT):
+    """Get all the clouds by the specified metadata url
+
+        :return: list of the clouds
+    """
+    import requests
+
+    try:
+        module_logger.debug('Start : Loading cloud metadata from the url specified by %s', metadata_url)
+        with requests.get(metadata_url, timeout=timeout) as meta_response:
+            arm_cloud_dict = meta_response.json()
+            cli_cloud_dict = _convert_arm_to_cli(arm_cloud_dict)
+            module_logger.debug('Finish : Loading cloud metadata from the url specified by %s', metadata_url)
+            return cli_cloud_dict
+    except Exception as ex:  # pylint: disable=broad-except
+        module_logger.warning("Error: Azure ML was unable to load cloud metadata from the url specified by %s. %s. "
+                        "This may be due to a misconfiguration of networking controls. Azure Machine Learning Python "
+                        "SDK requires outbound access to Azure Resource Manager. Please contact your networking team "
+                        "to configure outbound access to Azure Resource Manager on both Network Security Group and "
+                        "Firewall. For more details on required configurations, see "
+                        "https://docs.microsoft.com/azure/machine-learning/how-to-access-azureml-behind-firewall.",
+                        metadata_url, ex)
+        return {}
+
+def _convert_arm_to_cli(arm_cloud_metadata):
+    cli_cloud_metadata_dict = {}
+    if isinstance(arm_cloud_metadata, dict):
+        arm_cloud_metadata = [arm_cloud_metadata]
+    for cloud in arm_cloud_metadata:
+        try:
+            cli_cloud_metadata_dict[cloud['name']] = {
+                EndpointURLS.AZURE_PORTAL_ENDPOINT: cloud["portal"],
+                EndpointURLS.RESOURCE_MANAGER_ENDPOINT: cloud["resourceManager"],
+                EndpointURLS.ACTIVE_DIRECTORY_ENDPOINT: cloud["authentication"]["loginEndpoint"],
+                EndpointURLS.AML_RESOURCE_ID: cloud["resourceManager"],
```

Comment:
We should add a check somewhere in test/validations so that all the member of EndpointURLS should be overridden so that we can avoid any failures occurring in the new cloud because of missing overrides for any additional properties (that may be introduced in near future).
For e.g. A new endpoint may be added in under class EndpointURLS but the engineering may not be aware of other places to be uploaded to allow overrides of the new added properties, and we may encounter an issue during the deployment of the SDK in the overridden cloud.